xdem/coreg/base.py

```diff
@@ -379,7 +379,252 @@ def _preprocess_coreg_raster_input(
+            point_elev = to_be_aligned_elev
+            ref = "raster"
+        else:
+            raster_elev = to_be_aligned_elev
+            point_elev = reference_elev
+            ref = "point"
+
+        rst_elev, point_elev, inlier_mask, transform, crs = _preprocess_coreg_fit_raster_point(
+            raster_elev=raster_elev, point_elev=point_elev, inlier_mask=inlier_mask, transform=transform, crs=crs
+        )
+
+        if ref == "raster":
+            ref_elev = rst_elev
+            tba_elev = point_elev
+        else:
+            ref_elev = point_elev
+            tba_elev = rst_elev
+
+    # If both inputs are points, simply reproject to the same CRS
+    else:
+        ref_elev, tba_elev = _preprocess_coreg_fit_point_point(
+            reference_elev=reference_elev, to_be_aligned_elev=to_be_aligned_elev
+        )
+
+    return ref_elev, tba_elev, inlier_mask, transform, crs
+
+
+def _preprocess_coreg_apply(
+    elev: NDArrayf | MArrayf | RasterType | gpd.GeoDataFrame,
+    transform: rio.transform.Affine | None = None,
+    crs: rio.crs.CRS | None = None,
+) -> tuple[NDArrayf | gpd.GeoDataFrame, affine.Affine, rio.crs.CRS]:
+    """Pre-processing and checks of apply for any input."""
+
+    if not isinstance(elev, (np.ndarray, gu.Raster, gpd.GeoDataFrame)):
+        raise ValueError("Input elevation data should be a raster, an array or a geodataframe.")
+
+    # If input is geodataframe
+    if isinstance(elev, gpd.GeoDataFrame):
+        elev_out = elev
+        new_transform = None
+        new_crs = None
+
+    # If input is a raster or array
+    else:
+        # If input is raster
+        if isinstance(elev, gu.Raster):
+            if transform is not None:
+                warnings.warn(f"DEM of type {type(elev)} overrides the given 'transform'")
+            if crs is not None:
+                warnings.warn(f"DEM of type {type(elev)} overrides the given 'crs'")
+            new_transform = elev.transform
+            new_crs = elev.crs
+
+        # If input is an array
+        else:
+            if transform is None:
+                raise ValueError("'transform' must be given if DEM is array-like.")
+            if crs is None:
+                raise ValueError("'crs' must be given if DEM is array-like.")
+            new_transform = transform
+            new_crs = crs
+
+        # The array to provide the functions will be an ndarray with NaNs for masked out areas.
+        elev_out, elev_mask = get_array_and_mask(elev)
+
+        if np.all(elev_mask):
+            raise ValueError("'dem' had only NaNs")
+
+    return elev_out, new_transform, new_crs
+
+
+def _postprocess_coreg_apply_pts(
+    applied_elev: gpd.GeoDataFrame,
+) -> gpd.GeoDataFrame:
+    """Post-processing and checks of apply for point input."""
+
+    # TODO: Convert CRS back if the CRS did not match the one of the fit?
+    return applied_elev
+
+
+def _postprocess_coreg_apply_rst(
+    elev: NDArrayf | gu.Raster,
```

Comment:
What is "elev" here? 
It's a bit easier to follow the logic if you give that info in the docstring. I know these functions are not visible for the user and it's a bit annoying to fill the docstring, but for debugging, it will help for sure! 😄

Reply:
From what I understand, it's the original elevation data, used only to get the output grid?

Reply:
Exactly, I added an additional description in the function to explain this for our sake later on. 
I don't think we need a full parameter descriptions because except for this one, all other parameters are simply repeated from the main function that calls these non-public ones.